ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/BlockFetch.hs

```diff
@@ -103,24 +110,47 @@ startKeepAliveThread registry fetchClientRegistry peerId =
         atomically retry
 
 runBlockFetchClient ::
-     (Ord peer, IOLike m, MonadTime m, MonadTimer m, HasHeader blk, HasHeader (Header blk), ShowProxy blk)
-  => peer
-  -> FetchClientRegistry peer (Header blk) blk m
+     (IOLike m, MonadTime m, MonadTimer m, HasHeader blk, HasHeader (Header blk), ShowProxy blk)
+  => Tracer m (TraceEvent blk)
+  -> PeerId
+  -> StateViewTracers blk m
+  -> FetchClientRegistry PeerId (Header blk) blk m
   -> ControlMessageSTM m
   -> Channel m (AnyMessage (BlockFetch blk (Point blk)))
   -> m ()
-runBlockFetchClient peerId fetchClientRegistry controlMsgSTM channel = do
-    bracketFetchClient fetchClientRegistry ntnVersion isPipeliningEnabled peerId $ \clientCtx ->
-      void $ runPipelinedPeer nullTracer codecBlockFetchId channel
+runBlockFetchClient _tracer peerId StateViewTracers {svtPeerSimulatorResultsTracer} fetchClientRegistry controlMsgSTM channel = do
```

Comment:
What is the unused `_tracer` for?

Reply:
The unused `_tracer` and the `case fromException exn` are both here for the same thing: to allow tracing of component-related events, like what is done in `runChainSyncClient`:
```haskell
case fromException exn of
  Just CSClient.EmptyBucket ->
    traceWith tracer $ TraceChainSyncClientTerminationEvent peerId TraceTerminatedByLoP
  _ -> pure ()
```
For now, they aren't used for anything, that was just me pulling cables for potential future use.
Note that this happens in `runChainSyncServer`, `runBlockFetchClient`, and `runBlockFetchServer`.